### ISSUE TITLE
Question: Great difference of performance with hparams "transformer_librispeech_v1" and "transformer_librispeech_v2"

### ISSUE THREAD

**Issue by jitao-xu** [tensorflow/tensor2tensor#896]
### Description

I trained Librispeech clean small problem with Transformer model and I found that there was a great difference of performance when using the hyper-parameters set _transformer_librispeech_v1_ and _transformer_librispeech_v2_. I was using tensor2tensor 1.6.3

Command line _transformer_librispeech_v1_:
```
t2t-trainer --model=transformer \
--hparams_set=transformer_librispeech_v1 \
--problem=librispeech_clean_small \
--train_steps=30000 \
--eval_steps=3 \
--local_eval_frequency=10000 \
--data_dir=$DATA_DIR \
--output_dir=$TRAIN_DIR \
--worker_gpu=2
```
Evaluation result after 10000 steps:
```
dict for global step 10300: global_step = 10300, 
loss = 0.27501956, 
metrics-librispeech_clean_small/targets/accuracy = 0.919386,
metrics-librispeech_clean_small/targets/accuracy_per_sequence = 0.0034246575, 
metrics-librispeech_clean_small/targets/accuracy_top5 = 0.9834142, 
metrics-librispeech_clean_small/targets/edit_distance = 0.43903214, 
metrics-librispeech_clean_small/targets/neg_log_perplexity = -0.38055465
```

Command line _transformer_librispeech_v2_:
```
t2t-trainer --model=transformer \
--hparams_set=transformer_librispeech \
--problem=librispeech_clean_small \
--train_steps=500000 \
--eval_steps=3 \
--local_eval_frequency=10000 \
--data_dir=$DATA_DIR \
--output_dir=$TRAIN_DIR \
--worker_gpu=2
```
Evaluation result after 520000 steps:
```
dict for global step 520000: global_step = 520000, 
loss = 1.1357347, 
metrics-librispeech_clean_small/targets/accuracy = 0.6086176, 
metrics-librispeech_clean_small/targets/accuracy_per_sequence = 0.0, 
metrics-librispeech_clean_small/targets/accuracy_top5 = 0.8779174, 
metrics-librispeech_clean_small/targets/edit_distance = 0.3913824, 
metrics-librispeech_clean_small/targets/neg_log_perplexity = -1.2592698
```
The greatest difference between _transformer_librispeech_v1_ and _transformer_librispeech_v2_ is the batch size.
v1:
```
hparams.batch_size = 6000000
```

v2:
```
hparams.batch_size = 16
```

And when using a small batch size of 64 for _transformer_librispeech_v1_, the result became quiet similar with _transformer_librispeech_v2_  
```
dict for global step 10300: global_step = 300000, 
loss = 1.0917107, 
metrics-librispeech_clean_small/targets/accuracy = 0.5923423,
metrics-librispeech_clean_small/targets/accuracy_per_sequence = 0.0, 
metrics-librispeech_clean_small/targets/accuracy_top5 = 0.8761261, 
metrics-librispeech_clean_small/targets/edit_distance = 0.6058559, 
metrics-librispeech_clean_small/targets/neg_log_perplexity = -1.2856627
```

### Question
Do you know why using such a huge batch size in _transformer_librispeech_v1_ ?
And do you know why the batch size plays such a great role in influencing the performance?

**Comment by Qiaoxl** [comment]
@XUJitao 
There are many difference between v1 and v2 (actually transformer_librispeech_v2 is exactly transformer_librispeech):
hparam_set:                                    v1               v2
batch_size:	                               16	           6000000
hidden_size:                                   384              256
layer_prepostprocess_dropout:      0.2              0.1
max_length:                                   1240000     132000
max_input_seq_length:                  1550           1650
daisy_chain_variables:                   False           True
filter_size:                                       1536           1024
num_encoder_layers:                     6                 5
num_decoder_layers:                     4                 3
num_heads:                                    2                 4
ffn_layer:                                         conv_relu_conv        dense_relu_dense
relu_dropout:                                  0.2              0.1
conv_first_kernel:                           9                 3

**Comment by XUJitao** [comment]
@Qiaoxl Yeah that's true. In fact, I tried to modify many of them in v2 but I got the similar poor performance. Only changing the batch_size makes a better performance. I have changed batch_size of v2 to 600000 and the performance remained. I tried 6000000 and it got better. Similarly, I used batch_size = 64 for v1 and I got a poor performance, too.

**Comment by vergilus** [comment]
batch_size is the key 
check your  batch_size_means_token  setting, for ASR problems token size for parallel data is usually over 7k tokens per example

**Comment by veqtor** [comment]
How many GPU's do you need for 6,000,000 tokens per batch?
